enhancements/security/manual-cloud-credentials.md

```diff
@@ -0,0 +1,170 @@
+
+Each available update we process successfully is added to the reconciledAvailableUpdates list.
+
+### On-Going Reconciliation
+
+```bash
+$ oc adm credentials reconcile
+```
+
+This command would re-reconcile against the CredentialsRequests currently in the cluster. This is more of an on-going configuration management to ensure something doesn't change unexpectedly on us due to external factors. It would however support reducing permissions as a result of an additive only reconcile pre-upgrade.
+
+It should be noted that running this command after reconciling --available-updates, but before updating, would undo the pre-upgrade changes. As such this command should clear the reconciledAvailableUpdates list.
+
+
+## Open Questions [optional]
+
+ > 1. How can we handle net new CredentialsRequests in incoming release images?
+   * We could create the CredentialsRequest during the pre-upgrade reconciliation, but this does not guarantee that the target namespace for the component (where the Secret/ServiceAccount would be stored) exists.
+   * If we were to take the incoming CredentialsRequests, mind the credentials, and store in a temporary secret in the CCO namespace, we could later transfer it to the correct namespace when it exists.
+ > 1. Can the installer/CVO install manifests that contain status?
+   * If not we may need to hack around this by storing details on the minted credentials prior to installation as an annotation on the CredentialsRequest, and have the CCO reconcile it to it's final location in Status once we have a live object.
+   * Is it safe to edit a manifest from the release image prior to install?
+ > 1. Should the reconcile against current version command (i.e. no --available-updates flag) use the current CredentialsRequests in the cluster, or by extracting the current release image? The latter would allow us to prune net new CredentialsRequests processed for availableUpdates that were never installed.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+Copying net new CredentialsRequests from future upgrades, minting credentials, storing them in a temporary location, and copying them to their final location during the upgrade is particularly nebulous.
```

Comment:
Feels like this should simply be a new CRD to separate the minting of credentials (and saving them before we lose them) and delivering the creds/secret to its final destination.

Reply:
We would need to store in a secret regardless, is a new CRD still helpful in that case? I'm not immediately sure how to best apply that.

Reply:
yeah, the sensitive nature of the object contents does push us into using secrets directly...